src/Orchard.Web/Modules/Orchard.Setup/Services/SetupService.cs

```diff
@@ -154,6 +155,10 @@ public class SetupService : ISetupService {
                     schemaBuilder.AlterTable("Orchard_Framework_DataMigrationRecord",
                         table => table.AddUniqueConstraint("UC_DMR_DataMigrationClass_Version", "DataMigrationClass", "Version"));
 
+                    // Create the distributed lock record schema.
+                    var distributedLockSchemaBuilder = new DistributedLockSchemaBuilder(_shellSettings, schemaBuilder);

```

Comment:
I think we're talking about the same thing. It will work the way you say. My only concern is code duplication (a small amount of it). But if you think that's less of an evil than using `DistributedLockSchemaBuilder` here then I'll remove this line here and add back the `Migrations` class.